### PR TITLE
DOC: Add example for ``np.random.default_rng().geometric()``

### PR DESCRIPTION
**Pull request by linus-md**
Adds an example.

### CODE REVIEW

**Comment by ngoldbaum** [comment]
Hi! Thanks so much for trying to improve the numpy documentation!

It looks like you already have a lot of docs PRs open adding examples in docstrings across the codebase. I don’t want to discourage you from doing more of this, adding more examples is certainly a worthwhile effort. 

That said, opening lots of PRs in a short period of time without waiting for your other PRs to be reviewed and merged first can make it hard for reviewers to keep track of what you’re doing. I’d encourage you to slow down a bit and try not to have too many PRs in flight at once.

I’d also encourage you to try to see if there are any other PRs in the backlog for documentation improvements and see if you can look them over and try to provide constructive feedback. A good rule of thumb in open source contribution is that you should try to give feedback on as many PRs as you open, to share the load of reviewing PRs and maintaining the project.

**Comment by linus-md** [comment]
Hi, I am sorry I was creating to many PRs.  I didn't want to make your job harder. I will certainly slow down.

Thank you for for telling me about your rule of thumb. I am new to the open source community and thought only maintainers should give feedback. I will have a look at open PRs. 😄

**Comment by ngoldbaum** [comment]
> Thank you for for telling me about your rule of thumb. I am new to the open source community and thought only maintainers should give feedback. I will have a look at open PRs. 😄

Awesome! It may feel intimidating at first and you may make mistakes, we all do. Making sure documentation examples and the prose in docstrings is as clear as possible is very impactful work since users are often reading docstrings to answer their questions about numpy.

File: numpy/random/_generator.pyx [hunk 1]
```diff
@@ -454,6 +454,21 @@ cdef class Generator:
         >>> x-y
         0.08 # may vary
 
+        The distribution can be visualized with as follows:
```
Comment (charris) [inline numpy/random/_generator.pyx:457]: Omit "with".

**Comment by charris** [comment]
Hmm, this seems a bit much, involving a lot of dependencies, for a docstring. It might be better to have a gallery of distributions in the documentation somewhere, although Wikipedia does a fairly good job of that already. Might be better to find links.

**Comment by linus-md** [comment]
Another possibility could be to put the pyplot and scipy imports at the top and write something like: Using scipy and matplotlib before every plot. I could also close or add a link somewhere if you prefer that.

**Comment by rkern** [comment]
We do these histogram plots in many of the other distribution methods.

We have used scipy in a few of them (`vonmises`, `power`, and `zipf`), albeit with `# doctest: +SKIP` on all of the lines that depend on the scipy import or downstream results. I'm not entirely sure that we want to make that dependency more standard (since the PDFs _are_ all more conveniently available there), but it's doable.

**Comment by mattip** [comment]
As I commented elsewhere, I am not sure extensive examples belong in the docstrings. Especially for common distributions. Maybe we could expand the user guide with a section around random distributions, and somehow cue the CI to build that only when a documentation page changes.

**Comment by linus-md** [comment]
Given the discussion in #25349 I streamlined the example.

File: numpy/random/_generator.pyx [hunk 2]
```diff
@@ -454,6 +454,21 @@ cdef class Generator:
...
 
+        The distribution can be visualized as follows:
+
+        >>> import numpy as np
+        >>> import matplotlib.pyplot as plt
+        >>> import scipy.stats as stats
+        >>> l = 2
+        >>> rng = np.random.default_rng()
+        >>> sample = rng.exponential(scale=l, size=1000)
+        >>> count, bins, ignored = plt.hist(sample, 30, density=True)
```
Comment (eendebakpt) [inline numpy/random/_generator.pyx:465]:
```suggestion
        >>> count, bins, _ = plt.hist(sample, 30, density=True)
```
The _ is commonly used to ignore an unused output

File: numpy/random/_generator.pyx [hunk 3]
```diff
@@ -454,6 +454,21 @@ cdef class Generator:
         >>> x-y
         0.08 # may vary
 
+        The distribution can be visualized as follows:
+
+        >>> import numpy as np
+        >>> import matplotlib.pyplot as plt
+        >>> import scipy.stats as stats
+        >>> l = 2
```
Comment (eendebakpt) [inline numpy/random/_generator.pyx:462]: Rename `l` to `scale`?

File: numpy/random/_generator.pyx (hunk 2 above)
Comment (linus-md) [reply numpy/random/_generator.pyx:465]: This is a good idea and should be changed in a few other places as well!

File: numpy/random/_generator.pyx (hunk 3 above)
Comment (linus-md) [reply numpy/random/_generator.pyx]: @eendebakpt This change should actually be here #25356, in the literature the parameter is usually called lambda, which doesn't work obviously. In wikipedia they call it a "rate parameter". I could change it to rate. I would know what scale is supposed to mean.

File: numpy/random/_generator.pyx [hunk 4]
```diff
@@ -3412,19 +3412,34 @@ cdef class Generator:
...
+        Draw 10.000 values from the geometric distribution, with the 
+        probability of an individual success equal to ``p = 0.35``:
 
+        >>> p, size = 0.35, 10000
         >>> rng = np.random.default_rng()
-        >>> z = rng.geometric(p=0.35, size=10000)
+        >>> sample = rng.geometric(p=p, size=size)
 
-        How many trials succeeded after a single run?
+        What percentage of trials succeeded after a single run?
```
Comment (mattip) [inline numpy/random/_generator.pyx:3430]: If you want to use percentages you need to multiply the result by 100. Maybe proportion would be more appropriate?
```suggestion
        What proportion of trials succeeded after a single run?
```

**Comment by mattip** [comment]
Thanks @linus-md

**Pull request merged**